### DISCUSSION TITLE
[Q] How to add custom header?

### DISCUSSION THREAD

**Discussion by francsiswanto** [roadrunner-server/roadrunner#679] [Q&A] [answered]
Hi,

I am using RR2 and Laravel 8 with PHP 8.
My existing project have entry to put custom header from the start in index.php as additional information for user, like this:
```
define('LARAVEL_START', microtime(true));
header("PHP-Start: " . date("Y-m-d H:i:s", LARAVEL_START));
```
How to configure RR to expose the custom header.

Thank you.

**Comment by tarampampam** [comment, marked as answer]
Hi @francsiswanto!

You should not use [header()](https://www.php.net/manual/en/function.header.php) function for header setting - Middleware will be great for this. Something like that:

```php
namespace App\Http\Middleware;

class AppendStartedAtHeaderMiddleware
{
    /**
     * Handle an incoming request.
     *
     * @param  \Illuminate\Http\Request  $request
     * @param  \Closure  $next
     * @return mixed
     */
    public function handle($request, Closure $next)
    {
        /** @var \Illuminate\Http\Response $response */
        $response = $next($request);

        $response->header('PHP-Start', date("Y-m-d H:i:s", LARAVEL_START));

        return $response;
    }
}
```

> More docs about middlewares: https://laravel.com/docs/8.x/middleware#introduction

And put `define('LARAVEL_START', microtime(true));` into your worker before the loop starting.

> Have you seen the <https://github.com/spiral/roadrunner-laravel> package for roadrunner&laravel integration?

**Reply by francsiswanto** [reply]
Thank you @tarampampam, it work as expected. Yes I'm starting or trying to use https://github.com/spiral/roadrunner-laravel since Laravel Octane not work on windows platform even using RR. Thanks

**Reply by rustatian** [reply]
@francsiswanto Just curious, why laravel octane doesn't work on the Windows platform?

**Reply by francsiswanto** [reply]
@48d90782 I just don't know why, but that's the answer from their team (https://github.com/laravel/octane/issues/282), I guess they do not aware there are people that live only on windows platform like me :).

**Reply by wolfy-j** [reply]
Sounds like the relay on process forks to download RR binaries. :(

**Reply by francsiswanto** [reply]
Hi @tarampampam, just want to share my finding, $response->header() not working for BinaryFileResponse class so I change from  
```
$response->header('PHP-Start', date("Y-m-d H:i:s", LARAVEL_START));
```
into:
```
$response->headers->set('PHP-Start', date("Y-m-d H:i:s", LARAVEL_START));
```
Thanks

**Comment by francsiswanto** [comment]
Thank you @tarampampam, I will try it tomorrow.
